examples/real_function_space.py

```diff
@@ -145,17 +179,31 @@ def u_exact(x):
 pc.setType("lu")
 pc.setFactorSolverType("mumps")
 
-xh = dolfinx.fem.petsc.create_vector_block(L)
+if main_assembly:
+    xh = b.duplicate()
+else:
+    xh = dolfinx.fem.petsc.create_vector_block(L)
+
```

Comment:
Do we also need to destroy `xh` is we don't duplicate? What does `duplicate` do by the way? Is it creating a copy of the object (similar to a clone in Rust)?

Reply:
Destroyed now.
Creates a vector of similar type, but does not copy data: https://petsc.org/release/manualpages/Vec/VecDuplicate/

Reply:
Actually, it is destroyed below. We always need to destroy petsc objects from the python side, or call the garbage cleanup command in PETSc.